openshift/generate-manifests.sh

```diff
@@ -54,7 +54,13 @@ mkdir -p "${TMP_ROOT}/openshift"
 cp -a "${REPO_ROOT}/openshift/kustomize" "${TMP_ROOT}/openshift/kustomize"
 
 # Override OPENSHIFT-NAMESPACE to ${NAMESPACE}
-find "${TMP_ROOT}" -name "*.yaml" -exec sed -i "s/OPENSHIFT-NAMESPACE/${NAMESPACE}/g" {} \;
+if [[ "$(uname)" == "Darwin" ]]; then
+  # macOS
+  find "${TMP_ROOT}" -name "*.yaml" -exec sed -i '' "s/OPENSHIFT-NAMESPACE/${NAMESPACE}/g" {} +
+else
+  # Linux
+  find "${TMP_ROOT}" -name "*.yaml" -exec sed -i "s/OPENSHIFT-NAMESPACE/${NAMESPACE}/g" {} +
+fi
```

Comment:
Does the `macOS` variant of this work on linux hosts? (I think the answer is yes, but I don't know for sure).

If so, let's just change to a portable command that works on both platforms?

Reply:
We tried that and failed.
So, now I pushed the condition.

Reply:
I tested it on my macOS vs Linux, and it seemed to work? I swear we had another case like this... 
We could also add `-e` before the script.
I swear we've fixed this before without resorting to checking for `Darwin`...

Reply:
I tried checking older versions, and the same issue was present (likely related to the latest change in this line).
I also attempted using awk, but I wasn’t successful.

At this point, I think it might be reasonable to consider sticking with +1 for +e. I’m not sure how much effort we want to invest in removing the condition entirely.

Reply:
Maybe: https://github.com/openshift/release/blob/9712fce754de5b64f9287e3a5aea081ab29fcfde/tools/hack/golang/hack/lib/util/misc.sh#L121-L129

Reply:
```
function os::util::sed() {
	local sudo="${USE_SUDO:+sudo}"
	if LANG=C sed --help 2>&1 | grep -q "GNU sed"; then
		${sudo} sed -i'' "$@"
	else
		${sudo} sed -i '' "$@"
	fi
}
readonly -f os::util::sed

Reply:
Let me play with it for a sec.

Reply:
This worked for me with both macOS-native sed and gnu-sed.
```suggestion
find "${TMP_ROOT}" -name "*.yaml" -exec sed -i'.bak' "s/OPENSHIFT-NAMESPACE/${NAMESPACE}/g" {} \;
```

Reply:
Worked. Cool.

Reply:
But that leaves us with a `.bak` file which we should subseuqently `.gitignore` if not already there.

Reply:
Or, even better: append an `&& find . '*.bak' -exec rm {} \;`